### PR TITLE
feat(analytics)!: add missing stats

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### API Changes
* [x] Test the API endpoints which were added/changed.
* [x] Review the API documentation changes and confirm that it matches the actual functionality.
* [x] Check for breaking changes in the API and matching (conventional) commit message prefix.

#### Test cases
* [x] Review and run tests that were added/changed.
* [x] Suggest places that may benefit from test cases.

#### INX Changes
* [x] Run chronicle using an INX connection.

#### Database Changes
* [x] Review database queries for correctness/conciseness.
* [x] Ensure queries are supported by indexes if needed.
* [x] Check for breaking changes in the data model and matching (conventional) commit message prefix.

### CODE REVIEW

File: documentation/api/api-explorer.yml [hunk 1]
```diff
@@ -160,6 +160,57 @@ paths:
           $ref: "#/components/responses/NoResults"
         "500":
           $ref: "#/components/responses/InternalError"
+  /api/analytics/v2/ledger/richest-addresses:
```
[the remark this answers is not in the document]
Comment (Alexandcoats) [reply documentation/api/api-explorer.yml:163]: thanks

**Comment by Alex6323** [comment]
Was there a reason not to add integration tests for influx inserts (i.e. the analytics)? I would be fine with creating an issue since the PR is already huge, but it's one of the boxes to tick, so ...

File: src/bin/inx-chronicle/api/stardust/explorer/routes.rs [hunk 2]
```diff
@@ -206,3 +217,64 @@ async fn milestones(
...
+async fn resolve_ledger_index(database: &MongoDb, ledger_index: Option<MilestoneIndex>) -> ApiResult<MilestoneIndex> {
+    Ok(if let Some(ledger_index) = ledger_index {
+        ledger_index
+    } else {
+        database
+            .collection::<MilestoneCollection>()
+            .get_ledger_index()
+            .await?
+            .ok_or(ApiError::NoResults)?
+    })
```
Comment (Alex6323) [inline src/bin/inx-chronicle/api/stardust/explorer/routes.rs:279]: you can simplify this a bit by using `.unwap_or`

**Comment by Alexandcoats** [comment]
Perhaps we should wait until we settle on an influx lib to write integration tests. Also, you reminded me that I need to make influx optional.

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**